calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
```

Comment:
Can we make a new package `autodetection` for this code?  Maybe in `calico_node/autodetection`?

Keeping it separate will be better for adding alternative auto detection mechanisms down the road.